### PR TITLE
Changing docker client config attributes to Bazel labels

### PR DESCRIPTION
**Pull request by linzhp**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #2040

## What is the new behavior?

`toolchain_configure` only takes a Bazel label

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

Existing applications will have to change the `client_config` of their `toolchain_configure` rule to be a Bazel label.

## Other information
This is more correct than #2032 and #2041, but it introduces a breaking change.

### CODE REVIEW

File: toolchains/docker/toolchain.bzl [hunk 1]
```diff
@@ -68,8 +68,8 @@ docker_toolchain = rule(
         ),
         "client_config": attr.string(
             default = "",
-            doc = "A custom directory or a Bazel label for the docker client config.json. " +
-                  "If DOCKER_CONFIG is not specified, the value of the " +
+            doc = "A custom directory for the docker client config.json. If " +
```
Comment (uhthomas) [inline toolchains/docker/toolchain.bzl:71]: Would you agree the documentation for this attribute reads strangely? I don't really understand what it's trying to say. Might be good to update this?

**Review by uhthomas — COMMENTED** [review]
The PR and logic itself seems sound to me, I don't see any use-cases where an absolute path would be desirable. Despite this, it should still possible for the toolchain to use the client config specified in the home directory, or with the `DOCKER_CONFIG` environment variable.|

Pending response to the comments I left, I'm happy to approve this.

File: toolchains/docker/toolchain.bzl (hunk 1 above)
Comment (linzhp) [reply toolchains/docker/toolchain.bzl]: How about now?

Comment (uhthomas) [reply toolchains/docker/toolchain.bzl]: Much better, though the second to last sentence is missing an "If" at the beginning I believe.

**Review by uhthomas — APPROVED** [review]
Thank you for making those changes! LGTM

**Pull request merged**